### PR TITLE
adding changes required for installing Mellanox packages.

### PR DESCRIPTION
**Pull request by psairam**
This commit contains multiple things in order to support Mellanox changes with Nuage

* Installing Mellanox OFED & Redhat HotFix packages while patching.
* Adding Mlnx heat templates
* Documentation for Network Isolation.
* Support for 8.3.1-12 THT

### CODE REVIEW

File: image-patching/stopgap-script/README.md [hunk 1]
```diff
@@ -14,6 +14,8 @@ Copy the overcloud-full.qcow2 from undercloud-director /home/stack/images/ to th
 
 `cp overcloud-full.qcow2 overcloud-full-bk.qcow2`
 
+Untar the os-net-config.tar.gz provided by Mellanox and place it in the same location as overcloud-full.qcow2.
```
Comment (moshe010) [inline image-patching/stopgap-script/README.md]: This is just WA until all the patches of os-net-config will be merged and available by redhat. Do we need to document this here?

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py [hunk 2]
```diff
@@ -253,6 +254,22 @@ def copy_avrs_packages(image, avrs_baseurl, proxy_hostname = None, proxy_port =
     cmds_run(['rm -f nuage_avrs_packages'])
 
 
+#####
+# Installing Mellanox Packages
+#####
+
+def install_mellanox(image, workingDir):
+    # This is a temporary fix provided by Mellanox for os-net-config and will be removed in future releases
+    virt_copy('%s %s/os-net-config/os_net_config/* /usr/lib/python2.7/site-packages/os_net_config' % (image, workingDir))
```
Comment (moshe010) [inline image-patching/stopgap-script/nuage_overcloud_full_patch.py]: same as above

File: image-patching/stopgap-script/README.md (hunk 1 above)
Comment (psairam) [reply image-patching/stopgap-script/README.md]: Hi Moshe, I have updated this document with with "Note" https://github.com/nuagenetworks/nuage-ospdirector/pull/54/files#diff-2bec545a520361c8df6f55f470ed44b6R17. Please let me know how it looks. I am open to suggestions.

File: image-patching/stopgap-script/nuage_overcloud_full_patch.py (hunk 2 above)
Comment (psairam) [reply image-patching/stopgap-script/nuage_overcloud_full_patch.py]: Hi Moshe, I have updated the comment https://github.com/nuagenetworks/nuage-ospdirector/pull/54/files#diff-5678ac47e177f50d7f10526d27451edfR263. Please let me know how it looks. I am open to suggestions.

**Comment by psairam** [comment]
> is the firstboot script going to be on different commit?

I have updated this commit with firstboot script https://github.com/nuagenetworks/nuage-ospdirector/blob/be419035407f0906547db77944ce86a49d0950e6/tripleo-heat-templates-diff/common/diff_common_8.2.0-6.2#L127-L539.

**Pull request merged**